Refactoring .component to prevent weird wrapping issues

rhamilto opened the pull request:
@sg00dwin, @spadgett, @benjaminapetersen, @jwforres, @ajacobs21e, please review.

i made quite a few tweaks.  a few highlights:

* optimized existing responsive rules to be mobile first
* changed the .meta-data font size to 11px so it's the same size as .component-label so that they both sit on the same baseline
* changed the breakpoint where columns switch to rows from 768px to 992px so that the .meta-data only displays in the upper right of the box at medium and up sized screens
* fixed bug where .component h2 didn't truncate consistently (in most cases, the h2 would render full width)
* made each individual .port-mappings and .more-ports not wrap so all portions of each stay on the same line

Fixes #6189

spadgett commented:
Visually, the changes look really good to me.

spadgett commented:
Not sure if this is a new problem, but probably should get rid of the left margin when create route wraps this way.

<img width="376" alt="openshift_web_console" src="https://cloud.githubusercontent.com/assets/1167259/13124520/ddb51502-d58e-11e5-8686-ff80cca1750e.png">

spadgett commented on assets/app/styles/_components.less:
```diff
@@ -75,60 +75,50 @@
-          &.meta-data {
-            /* Left align for vertical layout at small sizes */
-            text-align: left;
+            margin-left: 5px;
```
Looks like this introduced the extra margin at mobile widths.

spadgett commented:
Just the one comment. The changes look really good otherwise. Good to see the wrapping problems fixed.

spadgett commented:
If you add "Fixes #6189" in the description, Github will close the issue when this merges. Generally prefer it there to the title where it looks like the PR number.

rhamilto commented:
Fixed the "Create route" issue and moved the issue number.  Should be good now, @spadgett.

![screen shot 2016-02-18 at 8 54 10 am](https://cloud.githubusercontent.com/assets/895728/13145367/2c4103cc-d61e-11e5-8bd9-fcfdecf404ff.PNG)

spadgett commented:
LGTM [merge]
